### ISSUE TITLE
dosdebug "u" without parameters and without prior "u" crashes dosemu

### ISSUE THREAD

**Issue by ecm-pushbx** [stsp/dosemu2#653]
**Describe the bug**
When starting dosdebug (with dosemu at the DOS command prompt, or with dosemu -H1 it happens too) and entering a "u" command, then dosemu crashes with the following:

```
$ dosemu -H1
ERROR: Fault in dosemu code, in_dpmi=0
ERROR: cpu exception in dosemu code outside of VM86()!
trapno: 0x0e  errorcode: 0x00000004  cr2: 0x00000000
eip: 0xb7d4c566  esp: 0xbfe1aad4  eflags: 0x00010283
cs: 0x0073  ds: 0x007b  es: 0x007b  ss: 0x007b
fs: 0x0000  gs: 0x0033

ERROR: Please install gdb, update dosemu from git, compile it with debug
info and make a bug report with the content of ~/.dosemu/boot.log at
https://github.com/stsp/dosemu2/issues
Please provide any additional info you can, like the test-cases,
URLs and all the rest that fits.

Terminated with signal 11
$ 
```

dosdebug quits due to this crash:

```
$ dosdebug


DOSEMU Debugger V0.6 connected
- type ? to get help on commands -
system state: emulated,stopped
AX=00c1  BX=0000  CX=0000  DX=0000  SI=0000  DI=0000  SP=fffe  BP=0000
DS=0040  ES=0000  FS=0000  GS=0000  FL=001b0202
CS:IP=f000:e05d       SS:SP=0000:fffe

f000:e05d B008             mov  al,08

system state: emulated,stopped
AX=00c1  BX=0000  CX=0000  DX=0000  SI=0000  DI=0000  SP=fffe  BP=0000
DS=0040  ES=0000  FS=0000  GS=0000  FL=001b0202
CS:IP=f000:e05d       SS:SP=0000:fffe

f000:e05d B008             mov  al,08
u

Dosemu process ended - quitting
$ 
```

**To Reproduce**
dosemu -H1, in another terminal dosdebug, in dosdebug "u". Compiled dosemu2 from git 990c48c98096

**Expected behavior**
Should disassemble at some default location (cs:ip ?) and not crash.

**Attach the log**
[boot.log](https://github.com/stsp/dosemu2/files/2237917/boot.log)

**Comment by ecm-pushbx** [comment]
"r" then "u" crashes too.

**Comment by ecm-pushbx** [comment]
"u cs:ip" doesn't crash, but typing "u" then crashes too.

**Comment by ecm-pushbx** [comment]
"u 0" doesn't crash (disassembles 0000:0000), and typing "u" then doesn't crash either (continues disassembly at 0000:0020).

**Comment by ecm-pushbx** [comment]
"u ip" gives "Invalid ADDR", doesn't crash.

**Comment by ecm-pushbx** [comment]
"u e05d" doesn't crash (disassembles at 0E05:000D), "u" then does crash.

**Comment by ecm-pushbx** [comment]
"u f000:e05d" crashes. 

```
$ dosemu -H1
ERROR: Fault in dosemu code, in_dpmi=0
ERROR: cpu exception in dosemu code outside of VM86()!
trapno: 0x0e  errorcode: 0x00000004  cr2: 0x00000000
eip: 0xb7dc7566  esp: 0xbfff7f64  eflags: 0x00010283
cs: 0x0073  ds: 0x007b  es: 0x007b  ss: 0x007b
fs: 0x0000  gs: 0x0033

ERROR: Please install gdb, update dosemu from git, compile it with debug
info and make a bug report with the content of ~/.dosemu/boot.log at
https://github.com/stsp/dosemu2/issues
Please provide any additional info you can, like the test-cases,
URLs and all the rest that fits.

Terminated with signal 11
$ 
```

[boot.log](https://github.com/stsp/dosemu2/files/2237921/boot.log)

**Comment by ecm-pushbx** [comment]
"u 0:0" doesn't crash (disassembles at 0000:0000).

**Comment by ecm-pushbx** [comment]
All crashes seem to be resolved with that (git 6d6eb7460dbb). However, "u ip" still replies with "Invalid ADDR". I didn't create an issue for that though because using a 16-bit register's value as linear address is not exactly useful nor a realistic use case.

**Comment by andrewbird** [comment]
Yes, I think we'll have to revisit how we deal with linear addresses to solve this. I favour requiring a prefix to determine if the address is to be interpreted as linear, but even then, as you suggest, we'll need to hint whether the value interpreted as an offset is to be relative to DS or CS.
